topic-operator/src/main/java/io/strimzi/operator/topic/K8sTopicWatcher.java

```diff
@@ -67,7 +72,16 @@ public void eventReceived(Action action, KafkaTopic kafkaTopic) {
     }
 
     @Override
-    public void onClose(KubernetesClientException e) {
+    public void onClose(KubernetesClientException exception) {
         LOGGER.debug("Closing {}", this);
+        Optional.ofNullable(exception)
+                .map(e -> {
+                    LOGGER.info("Exception received during watch", e);
+                    return exception;
+                })
+                .map(KubernetesClientException::getStatus)
+                .map(Status::getCode)
```

Comment:
You can combine these three into one. Also, is it really correct to only re-establish the watch on GONE, or should we do it whenever the exception is not null? IIRC that's what we do in CO and we should be consistent unless there's a good reason why not.